### PR TITLE
Functional tests - Add new test 'FO > checkout page > Select carrier'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Add new test 'FO > checkout page > Select carrier'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Nightly run => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/4175334795
| Fixed ticket?     | no
| Related PRs       | no
| Sponsor company   |

### CODE REVIEW

File: tests/UI/campaigns/functional/FO/11_checkout/03_shippingMethods/02_selectCarrier.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+      const addressesNumber = await checkoutPage.getNumberOfAddresses(page);
+      await expect(addressesNumber, 'The addresses number is not equal to 1!').to.equal(1);
+    });
+
+    it('should edit the created address', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'editCreatedAddress', baseContext);
+
+      await checkoutPage.clickOnEditAddress(page);
+
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/11_checkout/03_shippingMethods/02_selectCarrier.ts]:
```suggestion
```

File: tests/UI/pages/FO/checkout/index.ts [hunk 2]
```diff
@@ -509,8 +525,10 @@ class Checkout extends FOBasePage {
     await this.setValue(page, this.addressStepPostCodeInput, address.postalCode);
     await this.setValue(page, this.addressStepCityInput, address.city);
     await this.selectByVisibleText(page, this.addressStepCountrySelect, address.country);
-    await page.type(this.addressStepPhoneInput, address.phone, {delay: 50});
     await this.setValue(page, this.addressStepPhoneInput, address.phone);
+    if (await this.elementVisible(page, '#field-id_state', 1000)) {
```
Comment (Progi1984) [inline tests/UI/pages/FO/checkout/index.ts]: Could you move the selector, plz ?

**Review by Progi1984 — COMMENTED** [review]
Two small comments ;)

File: tests/UI/campaigns/functional/FO/11_checkout/03_shippingMethods/02_selectCarrier.ts (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/11_checkout/03_shippingMethods/02_selectCarrier.ts]: Done :)

File: tests/UI/pages/FO/checkout/index.ts (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/checkout/index.ts]: Yesss ;) Done :+1:

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**